src/Parser/RawBody.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+
+class RawBody implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    protected $buffer = '';
+    protected $contentLength;
+
+    public function __construct(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        $this->contentLength = $headers['content-length'];
```

Comment:
Sending out a `411` in case the `Content-Length` header is *required* sounds sane. However, I'm currently failing to see *where* this actually *is* required in this ticket?